server/handler/repo.go

```diff
@@ -52,6 +52,42 @@ func DelRepo(c web.C, w http.ResponseWriter, r *http.Request) {
 	var ctx = context.FromC(c)
 	var repo = ToRepo(c)
 
+	var rm = r.FormValue("remove") // using ?remove=true
+
+	// completely remove the repository from the database
+	if len(rm) != 0 {
+		var user = ToUser(c)
+		var remote = remote.Lookup(repo.Host)
+		if remote == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		// Request a new token and update
+		user_token, err := remote.GetToken(user)
+		if user_token != nil {
+			user.Access = user_token.AccessToken
+			user.Secret = user_token.RefreshToken
+			datastore.PutUser(ctx, user)
+		} else if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		// setup the post-commit hook with the remote system and
+		// if necessary, register the public key
+		var hook = fmt.Sprintf("%s/api/hook/%s/%s", httputil.GetURL(r), repo.Remote, repo.Token)
+		if err := remote.Deactivate(user, repo, hook); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
```

Comment:
I think we should proceed with deleting the repository from Drone even if it fails to delete the key or hook. We could log the error for debugging purposes, but it would proceed to the delete step.

The reason is that we cannot always control what is happening in GitHub, etc. There are edge cases I can think of where I want to delete the repository from Drone (perhaps as the Drone admin) but I don't have access to the repository in GitHub to remove the hooks.